sw/device/lib/testing/json/mem.h

```diff
@@ -0,0 +1,60 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_TESTING_JSON_MEM_H_
+#define OPENTITAN_SW_DEVICE_LIB_TESTING_JSON_MEM_H_
+
+#include "sw/device/lib/ujson/ujson_derive.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+// clang-format off
+
+#define MODULE_ID MAKE_MODULE_ID('j', 'm', 'h')
+
+#define STRUCT_MEM_READ32_REQ(field, string) \
+    field(address, uint32_t)
+UJSON_SERDE_STRUCT(MemRead32Req, mem_read32_req_t, STRUCT_MEM_READ32_REQ);
+
+#define STRUCT_MEM_READ32_RESP(field, string) \
```

Comment:
I don't know if it's worth changing that now but actually if you just want to return a uint32_t, you could directly call `ujson_serialize_uint32_t` and on the rust side `u32::recv`, i.e. you don't have to create a structure. But maybe it's cleaner with the structure.

Reply:
It actually makes a difference in what crosses the wire, though; those two options aren't the same.

Struct:
```json
{"value":1024}
```
u32:
```json
1024
```

While the struct _does_ take longer, the UART console isn't a dedicated channel, so those braces can help guard against some accidents (especially for responses like this one).

Reply:
That's true but my understanding of the protocol is that everything send starst with RESP_OK or RESP_ERR and is protected by CRC. I think the structure is also fine :)